### ISSUE TITLE
How to manually create Safari extension.

### ISSUE THREAD

**Issue by dustindauncey** [bitwarden/browser#889]
In an effort to mitigate the wait time noted in https://community.bitwarden.com/t/bitwarden-extension-in-safari-not-receiving-latest-updates-to-the-extension-do-they-still-apply-to-safari/4967, I was hoping to build the Safari extension manually from the source code, however I am not familiar with how to do it. I see on the readme instructions for Chrome and Firefox only. Does Safari need its own set of instructions added perhaps so users can manually update the extension instead of having to wait the near 2 months (due to the lack of a native app on macOS which would bypass that wait time via Safari App Extensions (different then Safari Extensions))?

**Comment by kspearrin** [comment]
Just run `npm run dist:safari` then you can side-load ./dist/Safari/bitwarden.safariextension into Safari. Please let me know if you find any issues in the current dev build as I am planning to submit it for review soon.

**Comment by dustindauncey** [comment]
I can't seem to do it successfully, I get errors in my build. But if you can build it successfully then I'm sure it's a local issue to my computer, as I'm new to building anything using npm, I may be missing some unlisted prerequisite or permission.

For what it's worth, the errors I see when running npm install (seems like I had to run that first then the one you noted above, as it also failed immediately if I ran just the npm run dist:safari without the npm install first) are as follows:

npm ERR! code ELIFECYCLE
npm ERR! errno 128
npm ERR! bitwarden-browser@0.0.0 postinstall: `npm run sub:init && gulp postinstall`
npm ERR! Exit status 128
npm ERR! 
npm ERR! Failed at the bitwarden-browser@0.0.0 postinstall script.

**Comment by kspearrin** [comment]
Do you have git installed? How did you checkout the codebase?

**Comment by dustindauncey** [comment]
I have git installed, yes, through the Apple Developer Tools. I was prompted to automatically install it when installing node.js. If I run “git” in terminal, I get the proper response to show it’s installed.

For checking out the codebase, I simply downloaded the assets using Github.com <http://github.com/> when looking at the Releases page for the project. I then used terminal to cd into the downloaded directory, then ran the commands to it can see the appropriate files. It then creates the new directory inside of it when I run the npm install command.
